docs/analytic_beam_tutorial.rst

```diff
@@ -0,0 +1,860 @@
+        sigma : float
+            Standard deviation in radians for the gaussian beam. Only one of sigma
+            and diameter should be set.
+        sigma_type : str
+            Either "efield" or "power" to indicate whether the sigma specifies the size of
+            the efield or power beam. Ignored if `sigma` is None.
+        diameter : float
+            Dish diameter in meters to use to define the size of the gaussian beam, by
+            matching the FWHM of the gaussian to the FWHM of an Airy disk. This will result
+            in a frequency dependent beam.  Only one of sigma and diameter should be set.
+        spectral_index : float
+            Option to scale the gaussian beam width as a power law with frequency. If set
+            to anything other than zero, the beam will be frequency dependent and the
+            `reference_frequency` must be set. Ignored if `sigma` is None.
+        reference_frequency : float
+            The reference frequency for the beam width power law, required if `sigma` is not
+            None and `spectral_index` is not zero. Ignored if `sigma` is None.
+        include_cross_pols : bool
+            Option to include the cross polarized beams (e.g. xy and yx or en and ne) for
+            the power beam.
+
+        """
+
+        sigma: float | None = None
+        sigma_type: Literal["efield", "power"] = "efield"
+        diameter: float | None = None
+        spectral_index: float = 0.0
+        reference_frequency: float = None
+
+        def __post_init__(self, include_cross_pols):
+            """
+            Post-initialization validation and conversions.
+
+            Parameters
+            ----------
+            include_cross_pols : bool
+                Option to include the cross polarized beams (e.g. xy and yx or en and ne)
+                for the power beam.
+
+            """
+            if (self.diameter is None and self.sigma is None) or (
+                self.diameter is not None and self.sigma is not None
+            ):
+                if self.diameter is None:
+                    raise ValueError("Either diameter or sigma must be set.")
+                else:
+                    raise ValueError("Only one of diameter or sigma can be set.")
+
+            if self.sigma is not None:
+                if self.sigma_type not in ["efield", "power"]:
+                    raise ValueError("sigma_type must be 'efield' or 'power'.")
+
+                if self.sigma_type == "power":
+                    self.sigma = np.sqrt(2) * self.sigma
+
+                if self.spectral_index != 0.0 and self.reference_frequency is None:
+                    raise ValueError(
+                        "reference_frequency must be set if `spectral_index` is not zero."
+                    )
+                if self.reference_frequency is None:
+                    self.reference_frequency = 1.0
+
+            super().__post_init__(include_cross_pols=include_cross_pols)
```

Comment:
Rather than making the user/developer have to worry about calling the `super().__post_init__`, perhaps we should define a `validate()` method on the base class, which is called in the base-class `__post_init__`. That hides the magic.

Reply:
I don't follow. I think if the user defines a `__post_init__` method the base-class `__post_init__` will not be run.

Reply:
I'm suggesting that in the base-class `__post_init__` we do:

```python
def __post_init__(self):
    ...what we have now...
    self.validate()

def validate(self):
    pass
```

Then subclasses can implement `validate` without having to call `super().__post_init__`

Reply:
Just pushed that change.